src/components/AuthContextProvider.js

```diff
@@ -0,0 +1,17 @@
+import { login } from '../utils/authenticate';
+import { createContext, useContext, useState } from 'react';
+
+export const AuthContext = createContext();
+export const useAuthContext = () => useContext(AuthContext);
+
+const AuthContextProvider = ({ children }) => {
+  const [jwt, setJwt] = useState();
+  const [userData, setUserData] = useState();
+  return (
+    <AuthContext.Provider value={{ jwt, setJwt, userData, setUserData, login }}>
```

Comment:
```suggestion
    <AuthContext.Provider value={{ jwt, userData, login }}>
```

We should handle the `setJwt` and `setUser` logic in the context itself. Why? Because if we want to reuse the login logic in several places (e.g. we have a login form, but also a login pop up overlay), we don't want to repeat this part in both places:

![image](https://user-images.githubusercontent.com/5387195/147760118-0ac04fc9-c8e3-4562-88cb-67168fe4ac45.png)

So, we should create a new function in the context here, e.g. `handleLogin` that does all this stuff, and provide that function from the context to the rest of the app

Also, `isLoggedIn` state can be a part of the AuthContext too